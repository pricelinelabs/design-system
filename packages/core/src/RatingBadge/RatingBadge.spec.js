import React from 'react'
import { render } from 'testing-library'

import { RatingBadge } from '..'

const text = '123'

describe('RatingBadge', () => {
  let consoleError
  beforeEach(() => {
    consoleError = console.error
    console.error = jest.fn()
  })
  afterEach(() => (console.error = consoleError))

  test('renders', () => {
    const { asFragment } = render(<RatingBadge>5</RatingBadge>)
    expect(asFragment()).toMatchSnapshot()
  })

  test('has correct background and text colors', () => {
    const color = 'secondary'

    const { getByText } = render(
      <RatingBadge color={color}>{text}</RatingBadge>
    )

    const badge = getByText(text)

    // Need to check the computed styles because .toHaveStyleRule(..) doesn't evaluate the functions
    const computedStyles = window.getComputedStyle(badge)
    expect(computedStyles.color).toEqual('rgb(255, 255, 255)')
    expect(computedStyles.backgroundColor).toEqual('rgb(0, 170, 0)')
  })

  test('renders with no bg and color', () => {
    const color = 'secondary'

    const { getByText } = render(
      <RatingBadge color={color}>{text}</RatingBadge>
    )

    const badge = getByText(text)

    const computedStyles = window.getComputedStyle(badge)
    expect(computedStyles.backgroundColor).toEqual('rgb(0, 170, 0)')
    expect(computedStyles.color).toEqual('rgb(255, 255, 255)')
  })

  test('renders default color with no props', () => {
    const { getByText } = render(<RatingBadge>{text}</RatingBadge>)

    const badge = getByText(text)

    const computedStyles = window.getComputedStyle(badge)
    expect(computedStyles.color).toEqual('rgb(255, 255, 255)')
    expect(computedStyles.backgroundColor).toEqual('rgb(246, 128, 19)')
  })
})
