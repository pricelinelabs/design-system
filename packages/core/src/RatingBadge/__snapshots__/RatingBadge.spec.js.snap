// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingBadge renders 1`] = `
<DocumentFragment>
  .c1 {
  padding-left: 8px;
  padding-right: 8px;
  background-color: #f68013;
  color: #fff;
}

.c2 {
  display: inline-block;
  line-height: 1.5;
  font-weight: 700;
  border-radius: 2px;
}

.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

<div
    class="c0"
  >
    <div
      class="c1 c2"
      color="alert"
      font-weight="bold"
    >
      5
    </div>
  </div>
</DocumentFragment>
`;
